src/suspense.js

```diff
@@ -0,0 +1,51 @@
+import { Component } from './component';
+import { createElement } from './create-element';
+
+// having a "custom class" here saves 50bytes gzipped
+export function Suspense(props) {}
+Suspense.prototype = new Component();
+
+/**
+ * @param {Promise} promise The thrown promise
+ */
+Suspense.prototype._childDidSuspend = function(promise) {
+	this.setState({ _loading: true });
+	const cb = () => { this.setState({ _loading: false }); };
+
+	// Suspense ignores errors thrown in Promises as this should be handled by user land code
+	promise.then(cb, cb);
+};
+
+Suspense.prototype.render = function(props, state) {
+	return state._loading ? props.fallback : props.children;
+};
+
+export function lazy(loader) {
+	let prom;
+	let component;
+	let error;
+
+	function Lazy(props) {
+		if (!prom) {
+			prom = loader();
+			prom.then(
+				(exports) => { component = exports.default; },
```

Comment:
Is this the right path here? Assuming that the `loader` would always return a _default_ export? That means we won't support named exports with lazy?

Reply:
React doesn't support this either, we could probably support it by introducing a second string param that defaults to `default` but is it really needed?

Reply:
This seems to be the preferred method (at least for [preact-async-route](https://github.com/prateekbh/preact-async-route#usage-example). Possibly change `(exports) => { component = exports.default; },` to `(exports) => { component = exports; },`. Then the user would be able to specify what the export is?

Reply:
No, this does not return to the user so the user can't alter the used export only if it's specified at initialization time.

Reply:
Right, that would lock in a pre-determined export.

Reply:
Ha, wasn't aware that React does the `default` only thing. I guess keeping it on par with React is what we should be doing, when it makes sense. I was thinking that maybe this is allowed:

```js
const Foo = lazy(() => import('./Foo').then(mod => mod.Foo));
```

Then we could basically support both of them?
```js
exports => { component = exports.default || exports },
```

This would allow the usage of default with named:
```js
const Foo = lazy(() => import('./Foo').then(mod => mod.Foo));
const Baz = lazy(() => import('./FooBaz));
```
But, feels like I might not be that familiar with `React.lazy`

Reply:
https://reactjs.org/docs/code-splitting.html#named-exports

Is the docs entry for it

Reply:
This is a great point! I think we should explore ways to use named exports after this PR has landed :+1: